### ISSUE TITLE
softmax used instead of log_softmax in metrics

### ISSUE THREAD

**Issue by deebuls** [torrvision/focal_calibration#14]
In the implementation of the ECE, ```softmax``` is used instead of ```log_softmax```.

some instances :
https://github.com/torrvision/focal_calibration/blob/475422a6754e20d25216158a874749c33e826e21/Metrics/metrics.py#L126 

(https://github.com/torrvision/focal_calibration/blob/475422a6754e20d25216158a874749c33e826e21/Metrics/metrics.py#L150) 

Is the observation correct ? because both in cross_entropy and focal_loss ```log_softmax``` is used after the logits. 

The probabilities will be different  in both case.

**Comment by vivkul** [comment]
Hi Deebul,

Thanks for going over the code. We need `softmax` to get the class probabilities output from the classifier.
`log_softmax` is required to get the log over probabilities which is needed to compute the loss (`cross_entropy` or `focal_loss`). I hope this helps.

Thanks!

**Comment by deebuls** [comment]
Yes thanks for the quick reply.  
I understand the answer I will close the issue.
